src/App.tsx

```diff
@@ -1,26 +1,87 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable no-console */
+/* eslint-disable max-len */
+import React, { useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { useTodos } from './hooks/useTodos';
+import { todosSlice } from './features/todos';
+import { useDispatch } from 'react-redux';
+import { filterFunc } from './utils/filterFunc';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+export enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+export const App: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const {
+    isTodosLoaded,
+    filterType,
+    query,
+    selectedTodo,
+    setIsTodosLoaded,
+    handleResetQuery,
+    setQuery,
+    anotherTodos,
+    filterRightNow,
+    queryRightNow,
+  } = useTodos();
+
+  useEffect(() => {
+    getTodos()
+      .then(current => {
+        dispatch(todosSlice.actions.add(current));
+      })
+      .catch(() => console.error('Error while getting todos!'))
+      .finally(() => setIsTodosLoaded(true));
+  }, []);
+
+  const filteredTodos: Todo[] = useMemo(() => {
+    return filterFunc(anotherTodos, {
+      filterType: filterRightNow,
+      query: queryRightNow,
+    });
+  }, [anotherTodos, filterRightNow, queryRightNow]);
```

Comment:
The `useMemo` hook is used to memoize `filteredTodos`. Ensure that `anotherTodos`, `filterRightNow`, and `queryRightNow` are correctly updated and reflect the current state when this memoization is recalculated. If these values are not updated correctly, it could lead to stale or incorrect data being displayed.